### PR TITLE
Always launch build under dev prompt

### PR DESCRIPTION
**Pull request by davkean**
Always launch build under dev prompt

To simplify finding VS tools and setting our environment, etc we always launch build under a VS command prompt.

### CODE REVIEW

**Comment by davkean** [comment]
Hold off reviewing this until I've figured out the kinks.

**Comment by davkean** [comment]
This is now ready for review.

**Comment by davkean** [comment]
Had to workaround: https://github.com/dotnet/project-system/issues/5177.

File: build/build.ps1 [hunk 1]
```diff
@@ -73,64 +73,6 @@ function GetVersion([string] $name) {
...
-  $vsWhereExe = GetVsWhereExe
-  $vsInstallDir = & $vsWhereExe -all -latest -prerelease -property installationPath    -requires Microsoft.Component.MSBuild -requires Microsoft.VisualStudio.Component.VSSDK -requires Microsoft.Net.Component.4.6.TargetingPack -requires Microsoft.VisualStudio.Component.Roslyn.Compiler
-  $vsVersion    = & $vsWhereExe -all -latest -prerelease -property installationVersion -requires Microsoft.Component.MSBuild -requires Microsoft.VisualStudio.Component.VSSDK -requires Microsoft.Net.Component.4.6.TargetingPack -requires Microsoft.VisualStudio.Component.Roslyn.Compiler
-
-  if (!(Test-Path $vsInstallDir)) {
-    throw "Failed to locate Visual Studio (exit code '$lastExitCode')."
-  }
-
-  if ($log) {
-    Write-Host "Using VS version $vsVersion in directory: $vsInstallDir"
```
Comment (drewnoakes) [inline build/build.ps1]: It'd still be useful to output this somewhere when `/diagnostic` is specified.

Comment (davkean) [reply build/build.ps1]: I had intentions of doing this via the command prompt logo itself which logs version. I'll do some follow up here around this.

**Pull request merged**